source/grid/manifold_lib.cc

```diff
@@ -373,6 +373,97 @@ get_tangent_vector (const Point<spacedim> &p1,
 
 
 
+template<int dim, int spacedim>
+Tensor<1, spacedim>
+SphericalManifold<dim, spacedim>::
+normal_vector (const typename Triangulation<dim, spacedim >::face_iterator &face,
+               const Point<spacedim>                                       &p) const
+{
+  // if the maximum deviation for the distance from the vertices to the center
+  // is less than 10% of the minimum distance to the first vertex, assume we can
+  // simply return p-center.
+  // otherwise, we compute the normal using get_normal_vector
+  constexpr unsigned int n_vertices = GeometryInfo<spacedim>::vertices_per_face;
+  std::array<double, n_vertices> distances_to_center;
+  std::array<double, n_vertices-1> distances_to_first_vertex;
+  distances_to_center[0] = (face->vertex(0)-center).norm_square();
+  for (unsigned int i=1; i<n_vertices; ++i)
+    {
+      distances_to_center[i] = (face->vertex(i)-center).norm_square();
+      distances_to_first_vertex[i-1] = (face->vertex(i)-face->vertex(0)).norm_square();
+    }
+  const auto minmax_distance = std::minmax_element(distances_to_center.begin(),
+                                                   distances_to_center.end());
+  const auto min_distance_to_first_vertex = std::min_element(distances_to_first_vertex.begin(),
+                                                             distances_to_first_vertex.end());
+
+  if (*minmax_distance.second-*minmax_distance.first<1.e-2 * *min_distance_to_first_vertex)
```

Comment:
Can we be more conservative here? I'd use a tolerance of 1e-6 or something like that for the following reason: We internally compute things to an accuracy of `1e-12` and this matches with the square I supposed to hold for the normals. That should catch all simple cases but avoid the ambiguity of using something different than what the base class promises. Or is there something I'm missing here?

Reply:
I read your comment 
> Would it make sense to add an assertion that the face is indeed on the surface, i.e., by checking that the distance of all vertices on the face is within say 10% of the distance of the first point?

as a proposal for a `1.e-2` tolerance. Since I don't really have a good intuition for a suitable tolerance here, I am happy with changing to `1.e-6`.

Reply:
> We internally compute things to an accuracy of 1e-12 and this matches with the square I supposed to hold for the normals.

Which line of code are you referring to?

Reply:
I agree that I went away from my initial idea with my previous comment, so yes, I'd choose that one.


 >> We internally compute things to an accuracy of 1e-12 and this matches with the square I supposed to hold for the normals.
>
> Which line of code are you referring to?

I had misremembered the internal accuracies, we use `1e-10` not `1e-12`. In that case, I'd say we should stick with `1e-10` as the square of the norms or `1e-5` as the actual deviation to not introduce another length scale.